frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx

```diff
@@ -48,24 +51,56 @@ const ItemBox: React.FC<ItemBoxProps> = ({ value, label }) => {
 };
 
 export type PoolShareProps = {
-  poolShare: PoolAmount;
-  assetOne: Asset;
-  assetTwo: Asset;
-  price: BigNumber;
-  revertPrice: BigNumber;
-  share: BigNumber;
+  pool: PoolConfig;
+  input: [Asset, Asset];
+  amounts: [BigNumber, BigNumber];
 } & BoxProps;
 
 export const PoolShare: FC<PoolShareProps> = ({
-  poolShare,
-  assetOne,
-  assetTwo,
-  price,
-  revertPrice,
-  share,
+  pool,
+  input,
+  amounts,
   ...rest
 }) => {
-  // TODO:Implement pool share
+  const isPoolsLoaded = useStore((store) => store.pools.isLoaded);
+  const [stats, setStats] = useState<GetStatsReturn>(null);
+  useEffect(() => {
+    if (isPoolsLoaded && pool) {
+      getStats(pool).then((result) => {
+        setStats(result);
+      });
+    }
+  }, [isPoolsLoaded, pool]);
+
+  const [assetLeft, assetRight] = input;
```

Comment:
NIT: Left/Right naming might be confusing. You can just use `assetOne` and `assetTwo`

Reply:
Same below for amounts

Reply:
Thanks for the review. applied.